plugins/modules/vmware_host_disk_info.py

```diff
@@ -0,0 +1,175 @@
+
+DOCUMENTATION = '''
+---
+module: vmware_host_disk_info
+short_description: Gathers information about disks attached to given ESXi host/s.
+description:
+- This module returns information about disks attached to given ESXi host/s
+- If I(cluster_name) is provided, then disk information about all hosts from the given cluster will be returned.
+- If I(esxi_hostname) is provided, then disk information about the given host system will be returned.
+author:
+- Matt Proud (@laidbackware)
+notes:
+- Tested on vSphere 6.7 and 7.0
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  cluster_name:
+    description:
+    - Name of the cluster from which the ESXi host belong to.
+    - If C(esxi_hostname) is not given, this parameter is required.
+    type: str
+  esxi_hostname:
+    description:
+    - ESXi hostname to gather information from.
+    - If C(cluster_name) is not given, this parameter is required.
+    type: str
+extends_documentation_fragment:
+- community.vmware.vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: Gather info about vmhbas of all ESXi Host in the given Cluster
+  community.vmware.vmware_host_disk_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: '{{ cluster_name }}'
+  delegate_to: localhost
+  register: cluster_host_vmhbas
+
+- name: Gather info about vmhbas of an ESXi Host
+  community.vmware.vmware_host_disk_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+  delegate_to: localhost
+  register: host_vmhbas
+'''
+
+RETURN = '''
+hosts_disk_info:
+  description: list of information for all disks attached to each ESXi host
+  returned: always
+  type: list
+  sample: >-
+    "192.168.0.182": [
+        {
+            "canonical_name": "naa.6000c296ed6217bd61df35622eb21a3a",
+            "capacity_mb": 4096,
+            "device_path": "/vmfs/devices/disks/naa.6000c296ed6217bd61df35622eb21a3a",
+            "device_type": "disk",
+            "device_ctd_list": [
+                "vmhba0:C0:T1:L0"
+            ],
+            "disk_uid": "key-vim.host.ScsiDisk-02000000006000c296ed6217bd61df35622eb21a3a566972747561",
+            "display_name": "Local VMware Disk (naa.6000c296ed6217bd61df35622eb21a3a)"
+        },
+        {
+            "canonical_name": "naa.6000c2968ad7142d93faae527fe8822b",
+            "capacity_mb": 204800,
+            "device_path": "/vmfs/devices/disks/naa.6000c2968ad7142d93faae527fe8822b",
+            "device_type": "disk",
+            "device_ctd_list": [
+                "vmhba0:C0:T3:L0"
+            ],
+            "disk_uid": "key-vim.host.ScsiDisk-02000000006000c2968ad7142d93faae527fe8822b566972747561",
+            "display_name": "Local VMware Disk (naa.6000c2968ad7142d93faae527fe8822b)"
+        },]
+'''
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
```

Comment:
```suggestion
```

Reply:
This module doesn't use vim, so the error occurs by CI.  
So please delete these lines.